skills/types/bool_file_test.go

```diff
@@ -0,0 +1,194 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package types_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/ubuntu-core/snappy/skills"
+	"github.com/ubuntu-core/snappy/skills/types"
+	"github.com/ubuntu-core/snappy/testutil"
+)
+
+func Test(t *testing.T) {
+	TestingT(t)
+}
+
+// BoolFileType
+
+type BoolFileTypeSuite struct {
+	testutil.BaseTest
+	t                skills.Type
+	gpioSkill        *skills.Skill
+	ledSkill         *skills.Skill
+	badPathSkill     *skills.Skill
+	missingPathSkill *skills.Skill
+	badTypeSkill     *skills.Skill
+	slot             *skills.Slot
+	badTypeSlot      *skills.Slot
+}
+
+var _ = Suite(&BoolFileTypeSuite{
+	t: &types.BoolFileType{},
+	gpioSkill: &skills.Skill{
+		Type: "bool-file",
+		Attrs: map[string]interface{}{
+			"path": "/sys/class/gpio/gpio13/value",
+		},
+	},
+	ledSkill: &skills.Skill{
+		Type: "bool-file",
+		Attrs: map[string]interface{}{
+			"path": "/sys/class/leds/input27::capslock/brightness",
+		},
+	},
+	missingPathSkill: &skills.Skill{
+		Type: "bool-file",
+	},
+	badPathSkill: &skills.Skill{
+		Type:  "bool-file",
+		Attrs: map[string]interface{}{"path": "path"},
+	},
+	badTypeSkill: &skills.Skill{
+		Type: "other-type",
+	},
+	slot: &skills.Slot{
+		Type: "bool-file",
+	},
+	badTypeSlot: &skills.Slot{
+		Type: "other-type",
+	},
+})
+
+func (s *BoolFileTypeSuite) TestName(c *C) {
+	c.Assert(s.t.Name(), Equals, "bool-file")
+}
+
+func (s *BoolFileTypeSuite) TestSanitizeSkill(c *C) {
+	// Both LED and GPIO skills are accepted
+	err := s.t.SanitizeSkill(s.ledSkill)
+	c.Assert(err, IsNil)
+	err = s.t.SanitizeSkill(s.gpioSkill)
+	c.Assert(err, IsNil)
+	// Skills without the "path" attribute are rejected.
+	err = s.t.SanitizeSkill(s.missingPathSkill)
+	c.Assert(err, ErrorMatches,
+		"bool-file must contain the path attribute")
+	// Skills with incorrect value of the "path" attribute are rejected.
+	err = s.t.SanitizeSkill(s.badPathSkill)
+	c.Assert(err, ErrorMatches,
+		"bool-file can only point at LED brightness or GPIO value")
+	// It is impossible to use "bool-file" type to sanitize skills of other types.
+	c.Assert(func() { s.t.SanitizeSkill(s.badTypeSkill) }, PanicMatches,
+		`skill is not of type "bool-file"`)
+}
+
+func (s *BoolFileTypeSuite) TestSanitizeSlot(c *C) {
+	err := s.t.SanitizeSlot(s.slot)
+	c.Assert(err, IsNil)
+	// It is impossible to use "bool-file" type to sanitize slots of other types.
+	c.Assert(func() { s.t.SanitizeSlot(s.badTypeSlot) }, PanicMatches,
+		`skill slot is not of type "bool-file"`)
+}
+
+func (s *BoolFileTypeSuite) TestSlotSecuritySnippet(c *C) {
```

Comment:
There's a lot happening in this test, maybe break it into 2: 1 testing error conditions and 1 testing the happy path?